### ISSUE TITLE
Fixup deprecated code identified by CIVET

### ISSUE THREAD

**Issue by cticenhour** [arfc/moltres#150]
This issue can be closed when deprecated MOOSE code and input file syntax [identified by CIVET testing](https://civet.inl.gov/job/674762/) is fixed. This can be checked in Moltres by using the `--error-deprecated` flag when performing local Moltres testing. 

Examples of the errors currently experienced are as follows:

## Deprecated D_name parameter
```
problems/033117_nts_temp_pre_parsed_mat.syntax: *** ERROR ***
problems/033117_nts_temp_pre_parsed_mat.syntax: 
problems/033117_nts_temp_pre_parsed_mat.syntax: 
problems/033117_nts_temp_pre_parsed_mat.syntax: Deprecated code:
problems/033117_nts_temp_pre_parsed_mat.syntax: The parameter D_name is deprecated.
problems/033117_nts_temp_pre_parsed_mat.syntax: This parameter has been renamed to 'diffusivity', which is more mnemonic and more conducive to passing a number literal
```
This also occurs in:
- `problems/2017_annals_pub_msre_compare.2group`
- `problems/2017_annals_pub_msre_compare.4group`
- `temp.temp`
- `problems/2017_annals_pub_msre_compare.auto_diff_rho`

## Changed addVariable API 

See [this MOOSE newsletter](https://mooseframework.inl.gov/newsletter/2019_08.html) and idaholab/moose#13769 for more info, and idaholab/moose#14061 for some module fixup examples that may prove useful (specifically [this commit](https://github.com/idaholab/moose/commit/b0d160c17f568ea78b96bbf985841cb0998401bc))

```
diracHX.dirachx: *** ERROR ***
diracHX.dirachx: 
diracHX.dirachx: 
diracHX.dirachx: Deprecated code:
diracHX.dirachx: Please use the addVariable(var_type, var_name, params) API instead
```
This also occurs in:
- `pre.pre`
- `nts.nts`
- `problems/2017_annals_pub_msre_compare.2group_eigen`
- `problems/2017_annals_pub_msre_compare.4group_eigen`

---
@katyhuff Apologies if I missed the mark on your preferred issue template! I took at look at CONTRIBUTING, but didn't see anything on issue creation.

**Comment by katyhuff** [comment]
Thanks a ton for letting us know @cticenhour ! 

@smpark7 I've listed you as the assignee to fix this. But, @yardasol @robfairh @ZoeRichter, any of you could assign it to yourselves instead, if preferred.

**Comment by cticenhour** [comment]
Happy to help! This caught my attention as I was looking through my CIVET dashboard. If the eventual assignee has questions, please feel free to let me or a member of the MOOSE team know!

**Comment by smpark7** [comment]
@yardasol @robfairh @ZoeRichter @LukeSeifert Feel free to work on this issue, along with #136 and #137 if you wish to.

I'm currently working on #152 and #151 so I won't be able to tackle this until next week or so.

**Comment by robfairh** [comment]
I'll see if I have some time for this today.

**Comment by smpark7** [comment]
Katy and I discussed this issue during our meeting today. She mentioned there's a possibility that the failed test in #151 is linked to the deprecated code.
@ZoeRichter @LukeSeifert If either of you want to give it a shot, we can do a quick video call to go through these issues. It's a good exercise if you're not yet quite familiar with some of moose/moltres' code. Otherwise, Roberto or I will tackle this today/tomorrow/this weekend.

**Comment by cticenhour** [comment]
Hi all! I noticed this email in my inbox, and took a look at your issue. We moved CIVET to some new build boxes, so CppUnit is now a module that can be loaded optionally by a testing recipe. Somehow this didn't get added to yours. You have two options:

(1) I can add `module load cppunit` to your test recipe on CIVET. 
(2) You can convert your `main.C` in `moltres/unit/src` to GoogleTest, which has been the new application standard for a while now (we include the header for it within MOOSE itself).

If (2) is your choice, you can take a look at the Stork template for that file here to get a start on converting things: 
https://github.com/idaholab/moose/blob/46f6f007f0dab21b8e3f31d4818da759cc7e4941/stork/unit/src/main.C

**Comment by smpark7** [comment]
@cticenhour Thanks for letting us know! You've saved us a ton of trouble. I'll take a shot at (2)

**Comment by robfairh** [comment]
@smpark7 can I get some guidance here? I am not very familiar with CIVET and googletest. So, I think I will focus on the failing tests. I understand that I need to fix the deprecated errors by modifying the syntax in the input files that are failing the tests. Is my understanding correct?

I ran ```./run_tests -j4 --error-deprecated``` and I got the following [output](https://github.com/arfc/moltres/files/6090202/moltres-test-error.txt) confirming the description of this issue.
